tools/checked-c-convert/ConstraintBuilder.cpp

```diff
@@ -57,51 +178,75 @@ class FunctionVisitor : public RecursiveASTVisitor<FunctionVisitor> {
   // In any of these cases, due to conditional expressions, the number of
   // variables on the RHS could be 0 or more. We just do the same rule
   // for each pair of q_i to q_j \forall j in variables_on_rhs.
-  void constrainAssign(uint32_t V, Expr *RHS) {
+  void constrainAssign( std::set<ConstraintVariable*> V, 
+                        Expr *RHS) {
     if (!RHS)
       return;
+
     Constraints &CS = Info.getConstraints();
-    std::set<uint32_t> W;
-    Info.getVariable(RHS, W, Context);
+    std::set<ConstraintVariable*> W = Info.getVariable(RHS, Context);
     if (W.size() > 0) {
       // Case 1.
-      for (const auto &I : W)
-        CS.addConstraint(
-            CS.createEq(CS.getOrCreateVar(V), CS.getOrCreateVar(I)));
+      // There are constraint variables for the RHS, so, use those over
+      // anything else we could infer. 
+      constrainEq(V, W, Info);
     } else {
       // Cases 2-4.
       if (RHS->isIntegerConstantExpr(*Context)) {
         // Case 2.
         if (!RHS->isNullPointerConstant(*Context,
-                                        Expr::NPC_ValueDependentIsNotNull))
-          CS.addConstraint(CS.createEq(CS.getOrCreateVar(V), CS.getWild()));
+          Expr::NPC_ValueDependentIsNotNull))
+          for (const auto &U : W)
```

Comment:
I don't understand why the iteration is over W.   I would think that it should be over V.